### PR TITLE
Skip stringify if json

### PR DESCRIPTION
**Pull request by barrachri**
Fixes https://github.com/jazzband/django-auditlog/issues/354

### CODE REVIEW

File: auditlog/diff.py [hunk 1]
```diff
@@ -67,6 +67,9 @@ def get_field_value(obj, field):
                 value = timezone.make_naive(value, timezone=timezone.utc)
         except ObjectDoesNotExist:
             value = field.default if field.default is not NOT_PROVIDED else None
+    elif isinstance(field, JSONField):
+        try:
+            value = field.to_python(getattr(obj, field.name, None))
```
Comment (Linkid) [inline auditlog/diff.py:72]: Exception should be handled

File: auditlog/diff.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from django.conf import settings
 from django.core.exceptions import ObjectDoesNotExist
-from django.db.models import NOT_PROVIDED, DateTimeField, Model
+from django.db.models import NOT_PROVIDED, DateTimeField, JSONField, Model
```
Comment (Linkid) [inline auditlog/diff.py:3]: You should prefer `from django_jsonfield_backport.models import JSONField` instead, for Django compat'.

**Review by Linkid — CHANGES_REQUESTED** [review]
Thanks for your contribution. I let you some comments.

File: auditlog/diff.py (hunk 1 above)
Comment (barrachri) [reply auditlog/diff.py:73]: Hi @Linkid, updated.

File: auditlog/diff.py (hunk 2 above)
Comment (barrachri) [reply auditlog/diff.py]: Also updated this one.

**Comment by barrachri** [comment]
Hi @Linkid, can you give the PR another check? Thanks!

**Review by Linkid — CHANGES_REQUESTED** [review]
Hi! Sorry about that, but with the PR #370, `django-jsonfield-backport` is removed. Then, you can use this line instead:
`from django.db.models import JSONField`

**Comment by alexsavio** [comment]
Sounds good to me. Thanks for the review @Linkid. When are you planning to make a release, though? Thanks a lot!

File: auditlog/diff.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 from django.db.models import NOT_PROVIDED, DateTimeField, Model
 from django.utils import timezone
 from django.utils.encoding import smart_str
+from django_jsonfield_backport.models import JSONField
```
Comment (barrachri) [inline auditlog/diff.py:6]:
```suggestion
from django.db.models import JSONField
```

**Comment by barrachri** [comment]
Thanks @hramezani. Do you think the added tests are enough?

**Comment by hramezani** [comment]
Thanks @barrachri for the tests. But I can see your tests are passing without the fix.
You need to add a test that shows the problem which your patch fixes it.

Also, Please add a changelog entry.

**Comment by barrachri** [comment]
Hi @hramezani you can try to run it again, while disabling the `elif isinstance(field, JSONField):` if branch.

I also added the changelog entry.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # Changes
 
+## 2.0.1
+
```
Comment (hramezani) [inline CHANGELOG.md:4]: let's remove the next version number. we will add it in the next release.
```suggestion
```

File: auditlog/diff.py [hunk 5]
```diff
@@ -67,6 +67,11 @@ def get_field_value(obj, field):
                 value = timezone.make_naive(value, timezone=timezone.utc)
         except ObjectDoesNotExist:
             value = field.default if field.default is not NOT_PROVIDED else None
+    elif isinstance(field, JSONField):
+        try:
+            value = field.to_python(getattr(obj, field.name, None))
+        except ObjectDoesNotExist:
```
[the remark this answers is not in the document]
Comment (barrachri) [reply auditlog/diff.py:73]: There's no test, _in general_, that covers the `except ObjectDoesNotExist:` part of get_field_value.

I am happy to add one though not entirely sure how to replicate that directly with the model (I believe we would need a migration to replicate it with save()).

But I can call model_instance_diff directly, with 2 instances and one has the attributed deleted.

**Review by hramezani — DISMISSED** [review]
Great work @barrachri :+1:

File: auditlog/diff.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (hramezani) [reply auditlog/diff.py:73]: BTW, it is a blocker because coverage fails and I can't merge the PR

**Comment by barrachri** [comment]
Thanks @hramezani, I added a new test and refactor the ObjectDoesNotExist try/except.

**Review by hramezani — APPROVED** [review]

**Pull request merged**